### PR TITLE
Fixes: #231 - Allow execution of custom code in zammad-init container

### LINKED ISSUE #231: [FR] customization settings persistent file
All customization settings (e.g. `Setting.set('storage_provider', "File")`) and [Advanced customization settings](https://docs.zammad.org/en/latest/admin/console/hidden-settings.html) shall be managed in a persistent file (e.g. `custom-settings.rb`) and applied on every deployment.

## cant use

* the [autoWizard](https://github.com/zammad/zammad-helm/blob/main/zammad/values.yaml#L301-L337), its only for inital setup
* [additional init containers](https://github.com/zammad/zammad-helm/blob/main/zammad/values.yaml#L375), cant set the needed dynamic env like `DATABASE_URL`, `REDIS_URL`  (see [zammad-railsserver](https://github.com/zammad/zammad-helm/blob/main/zammad/templates/statefulset.yaml#L274))

## Workaround

Use a persistent file (like `custom-settings.rb`) with kubectl or in your pipeline

`kubectl --namespace zammad exec -it zammad-0 --container=zammad-railsserver rails c < commands.rb`

### PR DESCRIPTION
**Pull request by mgruner**
#### Which issue this PR fixes

- fixes #231

#### Checklist

- [x] Chart Version bumped

### CODE REVIEW

File: zammad/Chart.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v2
 name: zammad
-version: 10.0.5
+version: 10.0.6
```
Comment (monotek) [inline zammad/Chart.yaml:3]: new feature should raise minor version.

File: zammad/templates/configmap-init.yaml [hunk 2]
```diff
@@ -36,4 +36,7 @@ data:
     if [ -n "${AUTOWIZARD_JSON}" ]; then
         echo "${AUTOWIZARD_JSON}" | base64 -d > /opt/zammad/var/auto_wizard.json
     fi
+    {{- if .Values.zammadConfig.initContainers.zammad.customCode }}
+      {{- .Values.zammadConfig.initContainers.zammad.customCode | nindent 4 }}
+    {{- end }}
```
Comment (monotek) [inline zammad/templates/configmap-init.yaml:41]: I woudl do it like this and name the variable "customInit".

```suggestion
      {{- with .Values.zammadConfig.initContainers.zammad.customInit }}
        {{- toYaml . | nindent 4 }}
      {{- end }}
```

File: zammad/values.yaml [hunk 3]
```diff
@@ -292,6 +292,8 @@ zammadConfig:
             - ALL
         readOnlyRootFilesystem: true
         privileged: false
+      # customCode: |
+      #   bundle exec rails runner '...'
```
Comment (monotek) [inline zammad/values.yaml:296]: i like an empty setting more. but no blocker...

```suggestion
      customInit: ""
        #  bundle exec rails runner '...'
```

**Review by monotek — CHANGES_REQUESTED** [review]

File: zammad/Chart.yaml (hunk 1 above)
Comment (monotek) [inline zammad/Chart.yaml:3]: new feature should raise minor version :)

```suggestion
version: 10.10
```

**Review by monotek — CHANGES_REQUESTED** [review]

**Comment by mgruner** [comment]
Thanks for the hints @monotek. I still need to learn a lot about k8s, helm and so on.

**Review by monotek — APPROVED** [review]

**Pull request merged**